src/main/java/org/graylog/plugins/threatintel/adapters/spamhaus/SpamhausEDROPDataAdapter.java

```diff
@@ -151,11 +152,18 @@ public LookupResult doGet(Object key) {
 
         // TODO potentially use a RangeMap here instead of doing linear searches all the time
         // We should be able to use ranges of the integer values of lowAddress and highAddress as keys
-        final Optional<Map.Entry<SubnetUtils.SubnetInfo, String>> match = subnets.get().values()
-                .stream()
-                .flatMap(list -> list.entrySet().stream())
-                .filter(entry -> entry.getKey().isInRange(ip))
-                .findFirst();
+        final Optional<Map.Entry<SubnetUtils.SubnetInfo, String>> match;
+        try {
+            match = subnets.get().values()
+                           .stream()
+                           .flatMap(list -> list.entrySet().stream())
+                           .filter(entry -> entry.getKey().isInRange(ip))
+                           .findFirst();
+        } catch (IllegalArgumentException e) {
+            // Gracefully handle the case when a blank or invalid IP is supplied.
+            LOG.error("[{}] is an invalid IP address. Lookup aborted. {}", ip, ExceptionUtils.getRootCauseMessage(e));
```

Comment:
Makes sense. Will do.

Reply:
@dennisoelkers and I discussed this. I am planning to change the error logging to `debug`, and continue to return an empty lookup result. This is an improvement on the current condition where the exception is thrown halting processing.

Lookup result currently does not support any ability to return an error to the UI. This would be a nice enhancement, so I am opening a separate issue. This can be considered for the Integrations roadmap at some point.

Reply:
@dennisoelkers I've changed the logging level for the invalid IP address case from error to debug. PR is ready for review.

We will be creating a separate Integrations plugin repository shortly. When that is done, I'll log an issue to add support for returning an error condition for failed lookups (should probably be part of the Integrations project).